libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultSpanScope.java

```diff
@@ -43,7 +50,8 @@ private DefaultSpanScope(Span span, SpanScope previousSpanScope, TracerContextSt
      */
     public static SpanScope create(Span span, TracerContextStorage<String, Span> tracerContextStorage) {
         final SpanScope beforeSpanScope = spanScopeThreadLocal.get();
-        SpanScope newSpanScope = new DefaultSpanScope(span, beforeSpanScope, tracerContextStorage);
+        final Span beforeSpan = tracerContextStorage.get(TracerContextStorage.CURRENT_SPAN);
```

Comment:
@Gaganjuneja thanks for the finding this issue out, the problem is basically related to split brain between `SpanScope` and `TracerContextStorage`, the former manages spans within local thread only whereas the later is used for inter-thread propagation.